src/utils.ts

```diff
@@ -81,6 +52,54 @@ export const isVideoJid = function (jid: string): boolean {
   return !!(jid && jid.match(/@conference/) && !isAcdJid(jid));
 };
 
+export type RetryPromise<T> = {
+  promise: Promise<T>;
+  cancel: (reason?: string | Error) => void;
+  complete: (value?: T) => void;
+  _id: string;
+};
+
+export function retryPromise<T> (
+  promiseFn: () => Promise<T>,
+  retryFn: (error?: Error | any) => boolean,
+  retryInterval: number = 15000
+): RetryPromise<T> {
```

Comment:
Main gist of `retryPromise`: 
1. Call the first param (which is a function that returns a `Promise`) until it resolves/rejects
2. Second param is a callback that is passed any errors received from the promise callback. If the call back returns `true`, it will retry the promise function
3. Retry on the interval passed in

It returns a promise that resolves/rejects when the promise callback does _OR_ if the returned `cancel() or complete()` functions are called which will short circuit the retry logic.